tests/std/tests/VSO_0191296_allocator_construct/test.cpp

```diff
@@ -217,44 +217,45 @@ struct emplace_argument {
-    class name {                                                                                                                 \
-    public:                 name(key) { }                                                                                        \
-    public:                 ~name() { }                                                                                          \
-    private: default_ctor   name(alloc_key) { }                                                                                  \
-    private: copy_ctor      name(alloc_key, name const&) { }                                                                     \
-    private: move_ctor      name(alloc_key, name&&) { }                                                                          \
-    private: emplace_ctor   name(alloc_key, emplace_argument&&) { }                                                              \
-    private: emplace_ctor   name(alloc_key, emplace_argument&&, emplace_argument&&) { }                                          \
-    private: emplace_ctor   name(alloc_key, emplace_argument&&, emplace_argument&&, emplace_argument&&) { }                      \
-    private: swappable      friend void swap(name&, name&) {}                                                                    \
-    private: copy_assign    name& operator=(name const&) { return *this; }                                                       \
-    private: move_assign    name& operator=(name&&) { return *this; }                                                            \
-    private: emplace_assign name& operator=(emplace_argument&&) { return *this; }                                                \
-    private: name() {}                                                                                                           \
-    private: name(const name&) {}                                                                                                \
+#define DEFINE_TYPE(name, def_ctor, copy_ctor, move_ctor, emp_ctor, swappable, copy_assign, move_assign, emp_assign) \
+    class name {                                                                                                     \
+    public:              name(key) { }                                                                               \
+    public:              ~name() { }                                                                                 \
+    private: def_ctor    name(alloc_key) { }                                                                         \
+    private: copy_ctor   name(alloc_key, name const&) { }                                                            \
+    private: move_ctor   name(alloc_key, name&&) { }                                                                 \
+    private: emp_ctor    name(alloc_key, emp_argument&&) { }                                                         \

```

Comment:
I believe this test is also affected (apologies for not commenting on all occurrences).

Reply:
Not your fault - I should have remembered there are two copies of this test. I'm doing five things at once tonight, and none of them terribly well.